### ISSUE TITLE
Board not Working

### ISSUE THREAD

**Issue by flixlix** [martijnvwezel/watermeter-esphome#8]
Hey there! First of all I would like to thank you for the effort with this board :)
I bought one and it arrived yesterday. Today I tried setting it up but unfortunately was unsuccessful...

# Problem/Motivation

> The board was never able to connect to my wi-fi even after entering the correct credentials in the webpage.

## Expected behavior

> Board connecting to wi-fi after entering credentials when connected to its AP.

## Actual behavior

> After entering the Credentials, the board entered into some error state and only logs the following when connected to it via USB:

```
invalid header: 0xffffffff
fffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
ESP-ROM:esp32c3-api1-20210207
Build:Feb  7 2021
rst:0x7 (TG0WDT_SYS_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
Saved PC:0x4004d1f8
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
```

## Steps to reproduce

> Open packaging -> Plugin USB-C Cable -> Wait for AP to appear -> Connect to AP -> Webpage opens automatically -> Select Network -> Enter Credentials -> (In my case) Error State: No Wireless connection to device possible anymore

## Proposed changes

> I'm not sure this means the board is bricked or not but I imagine a new board would fix the issue

**Comment by flixlix** [comment]
## Update

I was able to fix the above mentioned issue by entering the following:

```bash
python -m espefuse --port [MY_PORT] summary
```

But now I don't know how to get back the original firmware (or should I use the yaml and sh files in this repo?)

**Comment by martijnvwezel** [comment]
Very interesting error tbo. There is always a way to unbrick device. I still have to document that somewhere (short gpio pin 9 with ground or metal part on the esp32, casing, during boot)

you can reinstall  using the script in this repo

**Comment by martijnvwezel** [comment]
Try using the website button at the bottom (use chrome) or   
  
  


Clone the repo in linux env or so, and install using the script: `[production_installer.sh](https://github.com/martijnvwezel/watermeter-esphome/blob/main/production_installer.sh)` or upload using the button: [muino-water-meter-esp32.bin](https://github.com/martijnvwezel/watermeter-esphome/tree/gh-pages/muino-water-meter-esp32)

**Comment by martijnvwezel** [comment]
I make use of ESPHome, but I will try later this evening this weird issue. Maybe someone of Home-assistant or ESPHome users knows the issue.

**Comment by martijnvwezel** [comment]
I understand you want quick help, so if you want I can help you debugging in discord too with live chat

**Comment by martijnvwezel** [comment]
BTW insane slow shipping to Germany, to the USA it was in 6 days being received XD

**Comment by flixlix** [comment]
Hey there Martijn!
Thank you so much for the quick response and the very helpful support!
I will try to use the script in the website and report back later how it turned out :)

**Comment by martijnvwezel** [comment]
I heard from close sources of the ESPHome devs that today tomorrow they will rollout an update that is very important that actually fixes your issue

**Comment by flixlix** [comment]
Ooh that sounds promising!

Just tried uploading using the button at the bottom of [your page](https://watermeter.muino.nl/) but unfortunately it's stuck in "Preparing installation"... The console shows this, in case it helps:

<img width="1512" alt="Screenshot 2023-09-06 at 19 37 02" src="https://github.com/martijnvwezel/watermeter-esphome/assets/61006057/0a68e0db-1206-4a08-acbb-6f6cf5fefa7f">

Uploading from [this](https://esphome.github.io/esp-web-tools/) live demo is no issue.

Anyways, I'll wait for the next version and see if the issue gets fixed with the upload :)

**Comment by martijnED** [comment]
I will look at it later soery for the inconvenience it  causes. My watermeter stopped working after the last update.

**Comment by martijnvwezel** [comment]
I will look at it later sorry for the inconvenience it causes. My watermeter stopped working after the last update, so there actual some vague issues in ESPHome that are out of my control.

**Comment by martijnvwezel** [comment]
I fixed the website. The esphome repo with the template doesn't work anymore.

**Comment by martijnvwezel** [comment]
@flixlix  can you try uploading by website again? Please remove all previous created watermeters in home-assist under esphome and under devices->esphome

**Comment by flixlix** [comment]
Thanks for your support.
I was able to flash the board with a clean install and connect it to the network.
Unfortunately the issue persists, that the board does not connect to my wifi when powered by a usb cable plugged in to a power socket adapter. If that helps debug the issue, the board does connect to wifi when I open the logs and have it connected through usb on my laptop. Also, the board seems to be rebooting when connected to normal power, because the led turns off and back on after a few seconds

**Comment by martijnvwezel** [comment]
I already spammed one of the ESPHome devs, but they are not very helpfull. I will try to use a different board type maybe that helps not sure, though..
I created again a new issue about it with esphome. I saw yours too! Thankyou for that maybe that helps!

**Comment by flixlix** [comment]
I've edited the ESPHome code to look like this:

```yaml
esphome:
  name: test-device
  friendly_name: test-device
  includes:
    - my_custom_component.h

esp32:
  board: esp32-c3-devkitm-1
  framework:
    type: arduino

# Enable logging
logger:

# Enable Home Assistant API
api:

ota:
  password: "********************************"

wifi:
  ssid: !secret wifi_ssid
  password: !secret wifi_password

  # Enable fallback hotspot (captive portal) in case wifi connection fails
  ap:
    ssid: "Test-Device Fallback Hotspot"
    password: "************"

captive_portal:
    
web_server:
  port: 80

sensor:
  - platform: custom
    lambda: |-
      auto my_custom = new MyCustomSensor();
      App.register_component(my_custom);
      return {my_custom->water_liter_sensor, my_custom->sensa, my_custom->sensb, my_custom->sensc};

    sensors:
      - name: "water_liter_sensor"
        unit_of_measurement: L
        state_class: total_increasing
        device_class: water
        accuracy_decimals: 2
      - name: "sensa"
        accuracy_decimals: 2
      - name: "sensb"
        accuracy_decimals: 2
      - name: "sensc"
        accuracy_decimals: 2
```

and the top of the component to look like this:

```h
#define SENS_A 2
#define SENS_B 3
#define SENS_C 4

#define LED 7
#define LIGHT_SEN_ENABLE 6
```

This now fixes the issue, that it doesn't come online when powered externally (presumably because I changed the board type).

Although I didn't mount the sensor yet, the light is on and there are some random measurements going off (which I think is good).

Something that is still weird is that the logs are filled with this:

```txt


Time | level | Tag | Message
-- | -- | -- | --
01:04:08 | [D] | [api:102] | Accepted 192.168.68.100
01:04:08 | [D] | [api.connection:1031] | Home Assistant 2023.9.1 (192.168.68.100): Connected successfully
01:04:08 | [D] | [api.connection:171] | Home Assistant 2023.9.1 (192.168.68.100) requested disconnected
01:04:13 | [D] | [api:102] | Accepted 192.168.68.100
01:04:13 | [D] | [api.connection:1031] | Home Assistant 2023.9.1 (192.168.68.100): Connected successfully
01:04:13 | [D] | [api.connection:171] | Home Assistant 2023.9.1 (192.168.68.100) requested disconnected
01:04:18 | [D] | [api:102] | Accepted 192.168.68.100
01:04:18 | [D] | [api.connection:1031] | Home Assistant 2023.9.1 (192.168.68.100): Connected successfully
01:04:18 | [D] | [api.connection:171] | Home Assistant 2023.9.1 (192.168.68.100) requested disconnected
01:04:23 | [D] | [api:102] | Accepted 192.168.68.100
01:04:24 | [D] | [api.connection:1031] | Home Assistant 2023.9.1 (192.168.68.100): Connected successfully
01:04:24 | [D] | [api.connection:171] | Home Assistant 2023.9.1 (192.168.68.100) requested disconnected
01:04:29 | [D] | [api:102] | Accepted 192.168.68.100
01:04:29 | [D] | [api.connection:1031] | Home Assistant 2023.9.1 (192.168.68.100): Connected successfully
01:04:29 | [D] | [api.connection:171] | Home Assistant 2023.9.1 (192.168.68.100) requested disconnected
01:04:34 | [D] | [api:102] | Accepted 192.168.68.100
01:04:34 | [D] | [api.connection:1031] | Home Assistant 2023.9.1 (192.168.68.100): Connected successfully
01:04:34 | [D] | [api.connection:171] | Home Assistant 2023.9.1 (192.168.68.100) requested disconnected
01:04:39 | [D] | [api:102] | Accepted 192.168.68.100
01:04:39 | [D] | [api.connection:1031] | Home Assistant 2023.9.1 (192.168.68.100): Connected successfully
01:04:39 | [D] | [api.connection:171] | Home Assistant 2023.9.1 (192.168.68.100) requested disconnected
01:04:44 | [D] | [api:102] | Accepted 192.168.68.100
01:04:44 | [D] | [api.connection:1031] | Home Assistant 2023.9.1 (192.168.68.100): Connected successfully
01:04:44 | [D] | [api.connection:171] | Home Assistant 2023.9.1 (192.168.68.100) requested disconnected
01:04:50 | [D] | [api:102] | Accepted 192.168.68.100
01:04:50 | [D] | [api.connection:1031] | Home Assistant 2023.9.1 (192.168.68.100): Connected successfully
01:04:50 | [D] | [api.connection:171] | Home Assistant 2023.9.1 (192.168.68.100) requested disconnected
01:04:55 | [D] | [api:102] | Accepted 192.168.68.100
01:04:55 | [D] | [api.connection:1031] | Home Assistant 2023.9.1 (192.168.68.100): Connected successfully
01:04:55 | [D] | [api.connection:171] | Home Assistant 2023.9.1 (192.168.68.100) requested disconnected
01:04:59 | [D] | [api:102] | Accepted 192.168.68.100
01:04:59 | [W] | [api.connection:097] | 192.168.68.100: Reading failed: BAD_INDICATOR errno=11
01:05:00 | [D] | [api:102] | Accepted 192.168.68.100
01:05:00 | [D] | [api.connection:1031] | Home Assistant 2023.9.1 (192.168.68.100): Connected successfully
01:05:00 | [D] | [api.connection:171] | Home Assistant 2023.9.1 (192.168.68.100) requested disconnected
01:05:05 | [D] | [api:102] | Accepted 192.168.68.100
01:05:05 | [D] | [api.connection:1031] | Home Assistant 2023.9.1 (192.168.68.100): Connected successfully
01:05:05 | [D] | [api.connection:171] | Home Assistant 2023.9.1 (192.168.68.100) requested disconnected
01:05:10 | [D] | [api:102] | Accepted 192.168.68.100
01:05:10 | [D] | [api.connection:1031] | Home Assistant 2023.9.1 (192.168.68.100): Connected successfully
01:05:11 | [D] | [api.connection:171] | Home Assistant 2023.9.1 (192.168.68.100) requested disconnected
01:05:16 | [D] | [api:102] | Accepted 192.168.68.100
01:05:16 | [D] | [api.connection:1031] | Home Assistant 2023.9.1 (192.168.68.100): Connected successfully
01:05:16 | [D] | [api.connection:171] | Home Assistant 2023.9.1 (192.168.68.100) requested disconnected
01:05:21 | [D] | [api:102] | Accepted 192.168.68.100
01:05:21 | [D] | [api.connection:1031] | Home Assistant 2023.9.1 (192.168.68.100): Connected successfully
01:05:21 | [D] | [api.connection:171] | Home Assistant 2023.9.1 (192.168.68.100) requested disconnected
01:05:26 | [D] | [api:102] | Accepted 192.168.68.100
01:05:26 | [D] | [api.connection:1031] | Home Assistant 2023.9.1 (192.168.68.100): Connected successfully
01:05:26 | [D] | [api.connection:171] | Home Assistant 2023.9.1 (192.168.68.100) requested disconnected

```

I'm not sure if this is something to be worried about...

**Comment by martijnvwezel** [comment]
This 0xfffff is solved right?

**Comment by martijnvwezel** [comment]
btw you have to make sure previous watermeters are removed both under devices and esphome env

some issue in home assistant integration with esphome I heard

**Comment by martijnvwezel** [comment]
created in develop branch some pin defines for that board

**Comment by flixlix** [comment]
Great, thank you for the support!

Not sure if I should open a new issue for this, but I have the following situation:

The sensor is currently installed (with zipties because my water meter does not have any holes for them). I placed the circle outline over the spinning wheel in a way that I can still read the meter.

All sensors are reporting some values, but the water liter sensor is giving me odd values.

Is the value not supposed to be always rising? (accumulated water usage?)
The two spikes are correct, but the "standby" water usage is definitely 0, not 0.37
<img width="566" alt="Screenshot 2023-09-12 at 22 13 28" src="https://github.com/martijnvwezel/watermeter-esphome/assets/61006057/2e52ca77-a141-48cc-b225-f6b1293a33b9">

**Comment by flixlix** [comment]
Great, thank you for the support!

Not sure if I should open a new issue for this, but I have the following situation:

The sensor is currently installed (with zipties because my water meter does not have any holes for them). I placed the circle outline over the spinning wheel in a way that I can still read the meter.

All sensors are reporting some values, but the water liter sensor is giving me odd values.

Is the value not supposed to be always rising? (accumulated water usage?)
The two spikes are correct(in a sense that there was more water being consumed, the values are too low), but the "standby" water usage is definitely 0, not 0.37

<img width="566" alt="Screenshot 2023-09-12 at 22 13 28" src="https://github.com/martijnvwezel/watermeter-esphome/assets/61006057/2e52ca77-a141-48cc-b225-f6b1293a33b9">

**Comment by martijnvwezel** [comment]
This is what it should look like. This example has a reset of the wifi in it too. This up and down movement means/could mean the lichtsensors are nog correctly being positioned.
![image](https://github.com/martijnvwezel/watermeter-esphome/assets/4363788/3de5cd00-f7a6-4ed7-86a3-80f7c6284416)

**Comment by flixlix** [comment]
Have you tried / Can you verify that the pins I posted work with the new esp `esp32-c3-devkitm-1` board type?

**Comment by martijnvwezel** [comment]
see master branch, will check it, but first have to solve my server that broke down (sata connection, doesn't respond)

**Comment by martijnvwezel** [comment]
I think your pinout is wrong? Please see dev branch if that works! (currently raid setup still repairing estimated ~2days needed)
/```
/GPIO
#define D4 27
#define D5 14

// mapping to match other feathers and also in order
#define A0 26 
#define A1 25
#define A2 34
#define A3 39

#define SENS_A A0
#define SENS_B A1
#define SENS_C A2

#define LED D5
#define LIGHT_SEN_ENABLE D4
```

**Comment by martijnvwezel** [comment]
Could you give an update on your sensor. I hope this week I can make some time for it

**Comment by martijnvwezel** [comment]
BTW we are working on a implementation of Matter, I dislike the stability of ESPHome

**Comment by flixlix** [comment]
I updated the code with the pins you provided:

```C++
#define D4 27
#define D5 14

// mapping to match other feathers and also in order
#define A0 26
#define A1 25
#define A2 34
#define A3 39

#define SENS_A A0
#define SENS_B A1
#define SENS_C A2

#define LED D5
#define LIGHT_SEN_ENABLE D4
```

and the led doesn't light up anymore and the board became offline, it was working with the pins I provided but the values were off.

> BTW we are working on a implementation of Matter, I dislike the stability of ESPHome

That's interesting, because I have about 35 devices using ESPHome and have had 0 stability issues

**Comment by martijnED** [comment]
SO the moment you connect with serial usb it works right?

**Comment by flixlix** [comment]
No, unfortunately that still doesn't make it available.
Seems like with the pins you suggested it doesn't even boot...
Where did you get the pin numbers from?
I used the pinout schematic for the previous esp platform "seed..."

**Comment by martijnvwezel** [comment]
I will double check it soon, but here is the layout of the scheme

**Comment by martijnvwezel** [comment]
Please do the following:

``` yaml
logger:
  level: NONE
```

**Comment by flixlix** [comment]
With which code? The code in your dev branch or this code I provided?

>#define SENS_A 2
#define SENS_B 3
#define SENS_C 4
#define LED 7
#define LIGHT_SEN_ENABLE 6

Is it currently working in your end?

**Comment by martijnvwezel** [comment]
Thank you for helping me finding this issue! Yes you are correct, see dev branch: [watermeter_pinout](https://github.com/martijnvwezel/watermeter-esphome/blob/develop/my_custom_component.h)

**Comment by martijnvwezel** [comment]
I am not sure why the led pin is not turning on. Going to look at it soon..

**Comment by alx-a** [comment]
Thank you for the board, got it yesterday! I am experiencing issues somewhat similar to those above. Before I write up a proper report, some quick questions to move this forward:

1. Should I start a new issue?
2. Since logging  and then looking at logs would be my first start, I would like to make sure that enabling logging by changing the yaml config would not impact stability. Can I set it to logger level INFO ? currently:

logger:
  level: NONE

**Comment by martijnvwezel** [comment]
Currently, I am using Arduino  as my backbone for the project. I am going to remove that and only use yaml based implementation or esp lib only. There is an issue with Arduino and expressive librabry, if you want to use logging. Basically, after you restart and you do not open the serial port, the ESP crashing because of a watchdog that turns on.

**Comment by martijnvwezel** [comment]
@flixlix  how is your sensor now?

**Comment by alx-a** [comment]
In short, after ESPHome adoption (which includes reupload of their config and build on their end) the board bricked with the LED stopping to glow. 
I managed to get the board up with the docker build in the readme, updating the yaml config to my wifi.

The sensors are consistently reporting a minor .2 to .3 value when the meter water  wheel is not moving. 

I would appreciate if you could provide instructions or a flashable binary that you find works as expected. All I need is a means to communicate with HA, bypassing ESPHome is perfectly fine.

TLDR: ESPHome adoption bricks device. Meter can communicate with HA but is sending suspicious data. 
![weird_data](https://github.com/martijnvwezel/watermeter-esphome/assets/26557823/a54021e7-3432-4d6f-a109-b40fab22420c)

**Comment by martijnvwezel** [comment]
I will have a look into this issue soon. Please create a new ticket for this, it's a different matter. In the meantime, we will build some tools that

**Comment by martijnvwezel** [comment]
I close this issue because it now is in issue #16
